mozci/console/commands/check_backfills.py

```diff
@@ -121,47 +123,82 @@ def group_key(task):
             backfill_tasks = sorted(backfill_tasks, key=group_key)
             # Grouping ordered backfill tasks by their associated Treeherder symbol
             for th_symbol, value in groupby(backfill_tasks, group_key):
+                if th_symbol not in to_notify:
+                    to_notify[th_symbol] = {
+                        "newest_push": None,
+                        "backfill_tasks": set(),
+                    }
+
+                # make_push_objects returns the latest pushes in chronological order from oldest to newest
+                # We only need to store the newest Push that appeared for this Treeherder symbol
+                to_notify[th_symbol]["newest_push"] = push
                 tasks = list(value)
-                if all(task.state == "completed" for task in tasks):
-                    index_path = f"project.mozci.check-backfill.{environment}.{push.branch}.{push.rev}.{th_symbol}"
-                    try:
-                        find_task_id(
-                            index_path, root_url=COMMUNITY_TASKCLUSTER_ROOT_URL
-                        )
-                    except requests.exceptions.HTTPError:
-                        pass
-                    else:
-                        logger.debug(
-                            f"A notification was already sent for the backfill tasks associated to the Treeherder symbol {th_symbol}."
-                        )
-                        continue
+                # Storing all backfill tasks for this symbol across multiple pushes
+                to_notify[th_symbol]["backfill_tasks"].update(tasks)
```

Comment:
Nit: `to_notify[th_symbol]["backfill_tasks"].update(value)`

Reply:
(and maybe you can rename `value` to `tasks` or `tasks_iter` or something like that)

Reply:
done